src/Features/CSharp/Portable/MoveDeclarationNearReference/CSharpMoveDeclarationNearReferenceCodeRefactoringProvider.cs

```diff
@@ -54,5 +55,42 @@ protected override SyntaxToken GetIdentifierOfVariableDeclarator(VariableDeclara
 
             return true;
         }
+
+        protected override bool CanMoveToBlock(ILocalSymbol localSymbol, SyntaxNode currentBlock, SyntaxNode destinationBlock)
```

Comment:
I think it should be moved to helper.
@CyrusNajmabadi Could you please suggest a proper place for it?